support/conditions/conditions.go

```diff
@@ -0,0 +1,69 @@
+package conditions
+
+import (
+	hyperv1 "github.com/openshift/hypershift/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ExpectedHCConditions() map[hyperv1.ConditionType]metav1.ConditionStatus {
+	return map[hyperv1.ConditionType]metav1.ConditionStatus{
+		hyperv1.HostedClusterAvailable:               metav1.ConditionTrue,
+		hyperv1.InfrastructureReady:                  metav1.ConditionTrue,
+		hyperv1.KubeAPIServerAvailable:               metav1.ConditionTrue,
+		hyperv1.IgnitionEndpointAvailable:            metav1.ConditionTrue,
+		hyperv1.EtcdAvailable:                        metav1.ConditionTrue,
+		hyperv1.ValidReleaseInfo:                     metav1.ConditionTrue,
+		hyperv1.ValidHostedClusterConfiguration:      metav1.ConditionTrue,
+		hyperv1.SupportedHostedCluster:               metav1.ConditionTrue,
+		hyperv1.ClusterVersionSucceeding:             metav1.ConditionTrue,
+		hyperv1.ClusterVersionAvailable:              metav1.ConditionTrue,
+		hyperv1.ClusterVersionReleaseAccepted:        metav1.ConditionTrue,
+		hyperv1.ReconciliationActive:                 metav1.ConditionTrue,
+		hyperv1.ReconciliationSucceeded:              metav1.ConditionTrue,
+		hyperv1.ValidOIDCConfiguration:               metav1.ConditionTrue,
+		hyperv1.ValidAWSIdentityProvider:             metav1.ConditionTrue,
+		hyperv1.AWSDefaultSecurityGroupCreated:       metav1.ConditionTrue,
+		hyperv1.ValidHostedControlPlaneConfiguration: metav1.ConditionTrue,
+		hyperv1.ValidReleaseImage:                    metav1.ConditionTrue,
+		hyperv1.PlatformCredentialsFound:             metav1.ConditionTrue,
+		hyperv1.AWSEndpointAvailable:                 metav1.ConditionTrue,
+		hyperv1.AWSEndpointServiceAvailable:          metav1.ConditionTrue,
+
+		hyperv1.HostedClusterProgressing:  metav1.ConditionFalse,
+		hyperv1.HostedClusterDegraded:     metav1.ConditionFalse,
+		hyperv1.ClusterVersionFailing:     metav1.ConditionFalse,
+		hyperv1.ClusterVersionProgressing: metav1.ConditionFalse,
+
+		// conditions with no strong gurantees
+		//
+		// UnmanagedEtcdAvailable is only set if hcluster.Spec.Etcd.ManagementType == hyperv1.Unmanaged.
+		hyperv1.UnmanagedEtcdAvailable: metav1.ConditionTrue,
+		// ClusterVersionUpgradeable is not always guranteed to be true.
+		hyperv1.ClusterVersionUpgradeable: metav1.ConditionTrue,
+		// ExternalDNSReachable could be set to ConditionUnknown if external DNS is not configured or HC is private.
+		hyperv1.ExternalDNSReachable: metav1.ConditionTrue,
+		// ValidAWSKMSConfig could be set to ConditionUnknown if KMS key/role are not configured.
+		hyperv1.ValidAWSKMSConfig: metav1.ConditionTrue,
+	}
+}
+
+func ExpectedNodePoolConditions() map[string]corev1.ConditionStatus {
+	return map[string]corev1.ConditionStatus{
+		hyperv1.NodePoolReadyConditionType:                     corev1.ConditionTrue,
```

Comment:
I don't understand why we use corev1 condition here vs meta above. Seems odd to me that the ConditionStatus structs are duplicated.. ?